### PR TITLE
751: Create WMT user roles

### PR DESCRIPTION
**Pull request by ddebella**

### CODE REVIEW

File: migrations/app/20170912102542_app_users.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+exports.up = function (knex, Promise) {
+  return knex.schema.createTable('users', function (table) {
+    table.increments('id')
+    table.string('username').unique().notNullable()
+    table.string('forename')
```
Comment (caoimheb) [inline migrations/app/20170912102542_app_users.js]: Will we definitely be able to get `forename` and `surname`? And if so, what will they be used for?

File: seed/data/dev/01500_user_role.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+var tableName = 'user_role'
```
Comment (caoimheb) [inline seed/data/dev/01500_user_role.js]: Why not filename `01400_...`?

File: seed/data/ref/00300_roles.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+exports.seed = function (knex, Promise) {
+  // Deletes ALL existing entries
+  return knex(tableName).del()
+    .then(function () {
+      var sql = 'SET IDENTITY_INSERT app.' + tableName + ' ON;' +
+      insertStatement + '(1, \'Staff\')' +
+      insertStatement + '(2, \'Manager\')' +
+      insertStatement + '(3, \'System Admin\')' +
+      insertStatement + '(4, \'Data Admin\')'
+      return knex.schema
```
Comment (caoimheb) [inline seed/data/ref/00300_roles.js]: 👍

File: migrations/app/20170912102542_app_users.js (hunk 1 above)
Comment (ddebella) [reply migrations/app/20170912102542_app_users.js]: When I asked Tom this, his reply was that their existing system does contain forename an username and we should probably include it too. In our prototype there isn't a way to specify first and last names. One possible use would be for showing who you logged in as or in the case of workload point calculation the user who last updated the workloads points.

File: seed/data/dev/01500_user_role.js (hunk 2 above)
Comment (ddebella) [reply seed/data/dev/01500_user_role.js]: Must have missed that. I will change this.

**Pull request merged**